src/BinSkim.Driver/RoslynAnalyzer/ILOperations.cs

```diff
@@ -0,0 +1,485 @@
+    {
+        public Argument(IParameterSymbol parameter, IExpression value)
+        {
+            Parameter = parameter;
+            Value = value;
+        }
+
+        public IParameterSymbol Parameter { get; }
+        public IExpression Value { get; }
+
+        ArgumentKind IArgument.Kind => ArgumentKind.Positional;
+        IExpression IArgument.InConversion => null;
+        IExpression IArgument.OutConversion => null;
+    }
+
+    internal abstract class ReferenceExpression : Expression, IReferenceExpression
+    {
+    }
+
+    internal sealed class ArrayElementReferenceExpression : ReferenceExpression, IArrayElementReferenceExpression
+    {
+        public ArrayElementReferenceExpression(IExpression arrayReference, ImmutableArray<IExpression> indices)
+        {
+            ArrayReference = arrayReference;
+            Indices = indices;
+        }
+
+        public IExpression ArrayReference { get; }
+        public ImmutableArray<IExpression> Indices { get; }
+        public override ITypeSymbol ResultType => ((IArrayTypeSymbol)ArrayReference.ResultType).ElementType;
+        public override OperationKind Kind => OperationKind.ArrayElementReferenceExpression;
+    }
+
+    internal sealed class PointerIndirectionReferenceExpression : ReferenceExpression, IPointerIndirectionReferenceExpression
+    {
+        public PointerIndirectionReferenceExpression(IExpression pointer)
+        {
+            Pointer = pointer;
+        }
+
+        public IExpression Pointer { get; }
+        public override ITypeSymbol ResultType => ((IPointerTypeSymbol)Pointer.ResultType).PointedAtType;
+        public override OperationKind Kind => OperationKind.PointerIndirectionReferenceExpression;
+    }
+
+    internal class LocalReferenceExpression : ReferenceExpression, ILocalReferenceExpression
+    {
+        public LocalReferenceExpression(ILocalSymbol local)
+        {
+            Local = local;
+        }
+
+        public ILocalSymbol Local { get; }
+        public override ITypeSymbol ResultType => Local.Type;
+        public override OperationKind Kind => OperationKind.LocalReferenceExpression;
+    }
+
+    internal class ParameterReferenceExpression : ReferenceExpression, IParameterReferenceExpression
+    {
+        public ParameterReferenceExpression(IParameterSymbol parameter)
+        {
+            Parameter = parameter;
+        }
+
+        public IParameterSymbol Parameter { get; }
+        public override ITypeSymbol ResultType => Parameter.Type;
+        public override OperationKind Kind => OperationKind.ParameterReferenceExpression;
+    }
+
+    internal sealed class InstanceReferenceExpression : ParameterReferenceExpression, IInstanceReferenceExpression
+    {
+        public InstanceReferenceExpression(IMethodSymbol method)
+            : base(null) // FEEDBACK: There no public API to get the this parameter. 
+        {                // Do we need IInstanceReferenceExpression to be IParameterReferenceExpresison?
```

Comment:
I don't think the symbol is useful at the IOperation level. I was just commenting that I have no way to get it via public API and satisfy the current `IInstanceReferenceExpression : IParameterReferenceExpression`. I suggest making IInstanceReferenceExpression extend directly from IReferenceExpression.